### PR TITLE
Fix LaTeX drawer on split-filesystem systems

### PR DESCRIPTION
**Pull request by jmcelroy01**
### Summary

Corrects issue #8542 

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by jmcelroy01** [comment]
I'm unsure how to update a docstring (#2 in the pull request checklist).

File: qiskit/visualization/circuit_visualization.py [hunk 1]
```diff
@@ -488,7 +488,8 @@ def _latex_circuit_drawer(
         image = utils._trim(image)
         if filename:
             if filename.endswith(".pdf"):
-                os.rename(base + ".pdf", filename)
+                import shutil
```
Comment (jakelishman) [inline qiskit/visualization/circuit_visualization.py:491]: The import can go to the top of the file with the rest (I was just trying simplify the description of the change when I was asking you to try it out, sorry).

**Review by jakelishman — COMMENTED** [review]
The CI is just failing because of the code-style issue.  If you move the import to the top block (between `import os` and `import subprocess`) it should be happy again.

It would be good to add a release note to document the bug fix as well, but that will generally involve you setting up a proper development environment for Terra (see [our contributing guide](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md)), but if you've not got time, let me know and I can take care of it for you.

File: qiskit/visualization/circuit_visualization.py (hunk 1 above)
Comment (jmcelroy01) [reply qiskit/visualization/circuit_visualization.py]: Whoops, good point. I moved it.

**Comment by jakelishman** [comment]
Oh, I missed your comment before - don't worry about the docstring or adding additional tests for this change.  There's no documentation change necessary at all, and there's not really a sensible way we can test the LaTeX compilation on a split filesystem in CI anyway.

**Comment by jmcelroy01** [comment]
> It would be good to add a release note to document the bug fix as well, but that will generally involve you setting up a proper development environment for Terra (see [our contributing guide](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md)), but if you've not got time, let me know and I can take care of it for you.

Yeah, please. I did the commits on Github directly. Will look into doing this for next time :smile:

**Review by jakelishman — APPROVED** [review]
Looks good, thanks for the debugging and the quick responses!

**Pull request merged**